### PR TITLE
[Merged by Bors] - UCB: add block handler and generator

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

1st PR for #2906

## Changes

- add a temporary struct UCBlock. this will be renamed to Block in a later PR
- add block data handler when fetching from peers
- add block generator that generate a block from a list of proposals

## Test Plan

unit tests

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: blocks/reward.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	numProposals uint64
+	feesReward   uint64
+	layerReward  uint64
+	// smesher reward per proposal
+	totalRewardPer uint64
+	// the layer reward part of the smesher reward per proposal
+	layerRewardPer uint64
+}
+
+func (info *layerRewardsInfo) Fields() []log.LoggableField {
```
Comment (dshulyak) [inline blocks/reward.go]: is it a new code? should be using ObjectMarshaller interface if so

we should stop using Fields interface for logging, need to add a note that somewhere that it is deprecated

File: common/types/ucb.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	return bytes
+}
+
+// ID returns the BlockID.
+func (b *UCBlock) ID() BlockID {
+	return b.blockID
+}
+
+// Fields returns an array of LoggableFields for logging.
+func (b *UCBlock) Fields() []log.LoggableField {
```
Comment (dshulyak) [inline common/types/ucb.go]: should use ObjectMarshaler interface instead

**Review by dshulyak — APPROVED** [review]

File: blocks/reward.go (hunk 1 above)
Comment (countvonzero) [reply blocks/reward.go]: thanks for the reminder. i've dropped Fields here and for Ballot and Proposal too.

File: common/types/ucb.go (hunk 2 above)
Comment (countvonzero) [reply common/types/ucb.go]: done

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
> Build failed:
> 
> * [ci-stage2](https://github.com/spacemeshos/go-spacemesh/runs/4617008764?check_suite_focus=true)

turned out the system test relied on one of the proposal log field.

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**